src/dataBase/repository/books/bookRepo.go

```diff
@@ -40,27 +43,66 @@ func openDb() *sql.DB {
 	return db
 }
 
-func GetAll() ([]entity.Book, error){
+
+
+func GetByID(bookID int) (entity.Book, error) {
 	db:=openDb()
 
-	rows, err := db.Query("SELECT title FROM gotoboox.books")
+	rows, err := db.Query("SELECT title, description, popularity FROM gotoboox.books where id=$1", bookID)
 	if err != nil {
 		log.Fatal(err)
 	}
 	defer rows.Close()
 
-	books := []entity.Book{}
+	book := entity.Book{}
+
+	for rows.Next() {
+
+		book := new(entity.Book)
+		if err := rows.Scan(&book.Title);
+			err != nil {
+			log.Fatal(err)
+		}
+
+		//just for checking
+		fmt.Printf("%s\n%s\n%f\n", book.Title, book.Description, book.Popularity)
+
+	}
+	if err := rows.Err(); err != nil {
+		log.Fatal(err)
+	}
+	return book, nil
+
+
+
+
+}
+func GetAll() ([]interface{}, error){
+	db:=openDb()
```

Comment:
As far as i know, we dont need to open connection every time we execute a statement. Connection should be opened once